aten/src/ATen/core/ivalue.h

```diff
@@ -65,6 +65,44 @@ struct CAFFE2_API List : c10::intrusive_ptr_target {
   }
 };
 
+struct DictHash {
+  size_t operator()(const IValue& ivalue) const;
+};
+
+struct DictEqualTo {
+  bool operator()(const IValue& lhs, const IValue& rhs) const;
+};
+template <typename Key, typename Value>
+using DictUnorderedMap =
+  std::unordered_map<Key, Value, DictHash, DictEqualTo>;
+
+  template <typename Key, typename Value>
```

Comment:
DictHash and DictEqualTo make this only work with IValues. Why not just write a non-templated Dict? No reason for complexity that isn't useful.